### ISSUE TITLE
NullReferenceException when explicitly calling Close() on a Notification which has not yet been shown

### ISSUE THREAD

**Issue by praschl** [rafallopatka/ToastNotifications#60]
Hi,

I have a usecase, where i replace a Notification by another when the old is still shown and a status changes. To replace the notification I explicitly call `Close()` on the old Notification.

There is a second usecase where more about 15 Notifications are shown, leaving 5 Notifications hidden.

Now its possible, that a new Notification should replace one of the 5 hidden notifications. Since it has never been shown, its `Bind()` has never been called, and the `_closeAction` in `NotificationBase` is still null.

Long story short:
The NullReferenceException happens in NotificationBase:33, at the call `_closeAction.Invoke(this);` when I call `Close()` before `Bind()` has been called.

Edit:
I can workaround the Exception by checking if the Id of the notification is still 0. However, the notification will of course be shown (which is what I was trying to prevent).

**Comment by goyalsachin22** [comment]
were you able to clear the notifications using the below. I'm not able to.
 notifier.ClearMessages();

**Comment by praschl** [comment]
notifier.ClearMessages();
that works, but it clears all messages, and I would like to clear just the specific message.

**Comment by goyalsachin22** [comment]
notifier.ClearMessages(); not working for me (It's clearing the notifications). 


  public MainWindow()
        {
            InitializeComponent();
            this.meetingScreenHolder.Content = new Home();
            notifier = new Notifier(cfg =>
            {
                cfg.LifetimeSupervisor = new TimeAndCountBasedLifetimeSupervisor(TimeSpan.FromSeconds(5), MaximumNotificationCount.FromCount(15));
                cfg.PositionProvider = new PrimaryScreenPositionProvider(Corner.BottomRight, 10, 10);
                cfg.DisplayOptions.Width = 400;
                cfg.DisplayOptions.TopMost = true;
            });
            notifier.ShowCustomMessage("Disconnected", "Disconnected from companion app", Visibility.Collapsed);
            notifier.ShowCustomMessage("Licence Error", "Your Huddle license has expired. Please contact your BlueJeans administrator.");
          
                  }

  private void Button_Click(object sender, RoutedEventArgs e)
        {
            notifier.ClearMessages();
        }

**Comment by Lakritzator** [comment]
While you are at it, calling notifier.Dispose() also creates a NullReferenceException in TimeAndCountBasedLifetimeSupervisor.Dispose. I assume that the _notificationsPending.Clear(); is causing this.
A simple change to _notificationsPending?.Clear() should fix this.

**Comment by rafallopatka** [comment]
Hi thank you, both errors will be fixed in 2.3.3

**Comment by rafallopatka** [comment]
Version 2.3.3  is published now

**Comment by Lakritzator** [comment]
I can confirm that 2.3.3 fixes the dispose issue I reported. Thanks!!!!! 😃
